### PR TITLE
[Bug Fix] Fix to_homo when ndata has dgl.NTYPE or edata has dgl.ETYPE field.

### PR DESCRIPTION
**Pull request by classicsong**
## Description
#1719 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by BarclayII** [comment]
As in the discussion in #1719, the documentation of `to_homo` never intended to take the `dgl.NTYPE` features in heterogeneous graph as input.  So this PR is actually adding new behavior.

File: python/dgl/convert.py [hunk 1]
```diff
@@ -750,19 +750,33 @@ def to_homo(G):
     nids = []
     total_num_nodes = 0
 
+    ntid_dict = G.ndata[NTYPE] if len(G.ntypes) > 1 else {}
```
Comment (yzh119) [inline python/dgl/convert.py:753]: Is it equivalent to `G.ndata.get(NTYPE, {})`?

File: python/dgl/convert.py [hunk 2]
```diff
@@ -750,19 +750,33 @@ def to_homo(G):
...
     for ntype_id, ntype in enumerate(G.ntypes):
         num_nodes = G.number_of_nodes(ntype)
-        total_num_nodes += num_nodes
-        ntype_ids.append(F.full_1d(num_nodes, ntype_id, F.int64, F.cpu()))
+        if len(ntid_dict) > 0:
+            ntid = ntid_dict.get(ntype, None)
+            assert ntid is not None, 'if dgl.NTYPE field exist in heterograph, ' \
+                'all nodes should have this field.'
+            ntype_ids.append(ntid)
```
Comment (yzh119) [inline python/dgl/convert.py:760]: It seems `num_nodes` is not used in this branch and `ntype_ids` in L760 has a different length compared to `ntype_ids` in L763, how do you address the discrepancy?

**Comment by VoVAllen** [comment]
Is this fixed?

**Pull request closed without merging**